### DISCUSSION TITLE
Possible to only load the css when that component is in use?  Modern Module

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#5078] [Q&A]
Hi there, I am experimenting the Modern Module to build a component library. So far so good, however I realise that when I use the style.inject option it auto import the styles to the appropriate component which is good, but it loads the entire stylesheet for all components at once, is there are way to load on demand?

![image](https://github.com/web-infra-dev/modern.js/assets/52765572/901d12ac-ca73-4296-965e-eb66b894f0f4)

thanks :)

**Comment by 10Derozan** [comment]
Sorry, I miss the discussion.
I think you can use bundleless to load on demand. When buildType is bundleless, every component will have their own css.
`style.inject` will add it to the dom, and can not load on demand.
